src/CHIPBackend.cc

```diff
@@ -296,7 +298,9 @@ hipError_t CHIPModule::allocateDeviceVariablesNoLock(CHIPDevice *Device,
     queueVariableInfoShadowKernel(Queue, this, Var, &VarInfoBufD[I]);
     VarInfos.push_back(std::make_pair(Var, &VarInfoBufH[I]));
   }
-  Queue->memCopyAsync(VarInfoBufH.get(), VarInfoBufD, VarInfoBufSize);
+  auto Err =
+      Queue->memCopyAsync(VarInfoBufH.get(), VarInfoBufD, VarInfoBufSize);
+  assert(Err == hipSuccess);
```

Comment:
Asserts should be used liberally: https://llvm.org/docs/CodingStandards.html#assert-liberally. This assert, however, asks for a reason. I'll add one.

Reply:
Well in LLVM, asserts make sense. Here we have a hipError that we could return instead. Our error macros also print additional context/info. 
Why should we use an assert instead of `CHIPError` ?

Reply:
The assert failure here is considered unrecoverable situation (global variables won't work, most likely cause crashes). If the failure occurs we should fix whatever causes the failure.